packages/api/game/types.go

```diff
@@ -0,0 +1,140 @@
+package game
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+type BaseMessage struct {
```

Comment:
might be worth using a json rpc library like https://github.com/sourcegraph/jsonrpc2